singlem/package_creator.py

```diff
@@ -33,51 +33,72 @@ def create(self, **kwargs):
+                    num_seqs_unaligned += 1
+                    if s.id in tree_leaves:
+                        if s.id in found_sequence_names:
+                            raise Exception("Found duplicate sequence names in graftm unaligned"
+                                            " sequence fasta file. Currently this case is not handled,"
+                                            " sorry")
+                        SeqIO.write([s], filtered_aligned_tempfile, "fasta")
+                        found_sequence_names.add(s.id)
+                filtered_aligned_tempfile.flush()
 
-            if len(tree_leaves) != len(found_sequence_names):
-                for t in tree_leaves:
-                    if t not in found_sequence_names:
-                        raise Exception("Found some sequences that were in the tree but not the"
-                                        " unaligned sequences database e.g. %s. Something is"
-                                        " likely amiss with the input GraftM package" % t)
-                raise Exception("Programming error, shouldn't get here")
-            logging.info("All %i sequences found in tree extracted successfully from unaligned"
-                         " sequences fasta file, which originally had %i sequences" % (
-                             len(found_sequence_names), num_seqs_unaligned))
+                if len(tree_leaves) != len(found_sequence_names):
+                    for t in tree_leaves:
+                        if t not in found_sequence_names:
+                            raise Exception("Found some sequences that were in the tree but not the"
+                                            " unaligned sequences database e.g. %s. Something is"
+                                            " likely amiss with the input GraftM package" % t)
+                    raise Exception("Programming error, shouldn't get here")
+                logging.info("All %i sequences found in tree extracted successfully from unaligned"
+                            " sequences fasta file, which originally had %i sequences" % (
+                                len(found_sequence_names), num_seqs_unaligned))
+            else:
+                # GraftM package has no tree
+                num_seqs_unaligned = 0
+                filtered_aligned_tempfile = tempfile.NamedTemporaryFile(
+                        prefix='singlem_package_creator',
+                        suffix='.fasta',
+                        mode='w')
+                # okay for now, but copying the file would do just as well
+                for s in SeqIO.parse(gpkg.unaligned_sequence_database_path(), "fasta"):
+                    num_seqs_unaligned += 1
+                    SeqIO.write([s], filtered_aligned_tempfile, "fasta")
+                filtered_aligned_tempfile.flush()
+                logging.info("All %i sequences found in tree extracted successfully from unaligned"
```

Comment:
ah sorry this log message also says it was from the tree, which is technically incorrect. Just say unaligned db was copied directly.